src/main/java/mimon/Mimon.java

```diff
@@ -0,0 +1,204 @@
+        } else if (userInput.startsWith("mark ")) {
+            markTask(userInput, true);
+        } else if (userInput.startsWith("unmark ")) {
+            markTask(userInput, false);
+        } else if (userInput.startsWith("delete ")) {
+            deleteTask(userInput);
+        } else if (userInput.startsWith("todo")) {
+            if (userInput.trim().equals("todo")) {
+                throw new MimonException("The description of a todo must have some information.");
+            }
+            addTodoTask(userInput.substring(5).trim());
+        } else if (userInput.startsWith("deadline")) {
+            if (userInput.trim().equals("deadline")) {
+                throw new MimonException("The description of a deadline should be meaningful.");
+            }
+            addDeadlineTask(userInput.substring(9).trim());
+        } else if (userInput.startsWith("event")) {
+            if (userInput.trim().equals("event")) {
+                throw new MimonException("The description of an event should include the date!.");
+            }
+            addEventTask(userInput.substring(6).trim());
+        } else {
+            throw new MimonException("I don't recognize that command. Please enter a valid one.");
+        }
+    }
+
+    private static void printWelcomeMessage() {
+        System.out.println(DASHED_LINE);
+        System.out.println("Hello! I'm Mimon");
+        System.out.println("What can I do for you?");
+        System.out.println(DASHED_LINE);
+    }
+
+    private static void printExitMessage() {
+        System.out.println(DASHED_LINE);
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println(DASHED_LINE);
+    }
+
+    private static void listTasks() {
+        System.out.println(DASHED_LINE);
+        System.out.println("Here are the tasks in your list:");
+        if (tasks.isEmpty()) {
+            System.out.println("No tasks added yet.");
+        } else {
+            for (int i = 0; i < tasks.size(); i++) {
+                System.out.println((i + 1) + ". " + tasks.get(i));
+            }
+        }
+        System.out.println(DASHED_LINE);
+    }
+
+    private static void markTask(String userInput, boolean isDone) throws MimonException {
+        try {
+            int taskIndex = Integer.parseInt(userInput.split(" ")[1]) - 1;
+            if (taskIndex >= 0 && taskIndex < tasks.size()) {
+                if (isDone) {
+                    tasks.get(taskIndex).markAsDone();
+                    System.out.println(DASHED_LINE);
+                    System.out.println("Nice! I've marked this task as done:");
+                } else {
+                    tasks.get(taskIndex).markAsNotDone();
+                    System.out.println(DASHED_LINE);
+                    System.out.println("OK, I've marked this task as not done yet:");
+                }
+                System.out.println("  " + tasks.get(taskIndex));
+                System.out.println(DASHED_LINE);
```

Comment:
Avoid deep nesting. This is ~3 levels of indenation. You can consider refactoring the internals of this method into another one.